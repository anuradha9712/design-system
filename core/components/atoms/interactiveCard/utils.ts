export const removeItem = (targetItem: any, arr = []) => {
  return arr.filter((item: any) => item !== targetItem);
};
