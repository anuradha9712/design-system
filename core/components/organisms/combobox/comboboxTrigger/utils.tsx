import React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  // setFocusedOption?: React.Dispatch<React.SetStateAction<Element | undefined>>,
  setHighlightFirstItem?: any,
  setHighlightLastItem?: any
) => {
  console.log('handle key down e.key', event.key);

  switch (event.key) {
    case 'ArrowUp':
      setOpenPopover?.(true);
      setHighlightLastItem(true);
      // focusListItem('up', setFocusedOption);
      // requestAnimationFrame(() => focusListItem('up', setFocusedOption));
      break;
    case 'ArrowDown':
      setOpenPopover?.(true);
      setHighlightFirstItem(true);

      // requestAnimationFrame(() => focusListItem('down', setFocusedOption));
      break;
    case 'Escape':
      setOpenPopover?.(false);
      break;
    default:
      break;
  }
};

export const focusListItem = (
  position: string,
  setFocusedOption?: React.Dispatch<React.SetStateAction<Element | undefined>>
) => {
  const listItems = document.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let targetOption;
  // if (position === 'down') {
  //   targetOption = listItems[0];
  //   (listItems[0] as HTMLElement)?.focus();
  // } else {
  //   (listItems[listItems.length - 1] as HTMLElement)?.focus();
  // }

  if (position === 'down') {
    console.log('rrr down', listItems);
    targetOption = listItems[0];
  } else {
    targetOption = listItems[listItems.length - 1];
  }
  console.log('rrrtargetOption', targetOption);
  (targetOption as HTMLElement)?.focus();
  targetOption?.scrollIntoView({ block: 'center' });
  setFocusedOption && setFocusedOption(targetOption);
};
