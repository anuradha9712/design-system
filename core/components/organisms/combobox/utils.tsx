import React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<Element | undefined>>,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  inputTriggerRef?: any
) => {
  switch (event.key) {
    case 'ArrowUp':
      event.preventDefault();
      navigateOptions('up', focusedOption, setFocusedOption);
      break;
    case 'ArrowDown':
      event.preventDefault();
      navigateOptions('down', focusedOption, setFocusedOption);
      break;
    case 'Enter':
      handleEnterKey(focusedOption);
      break;
    case 'Escape':
      setOpenPopover?.(false);
      console.log('inputTriggerRef.current', inputTriggerRef.current);
      inputTriggerRef.current.focus();
      break;
    default:
      break;
  }
};

const handleEnterKey = (focusedOption: Element | undefined) => {
  (focusedOption as HTMLElement)?.click();
};

const navigateOptions = (
  direction: string,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<Element | undefined>>
) => {
  const listItems = document.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let index = Array.from(listItems).findIndex((item) => {
    return item == focusedOption;
  });

  if (index === -1) {
    index = direction === 'up' ? listItems.length - 1 : 0;
  } else {
    index = direction === 'up' ? (index - 1 + listItems.length) % listItems.length : (index + 1) % listItems.length;
  }

  const targetOption = listItems[index];
  console.log('targetOptiontargetOption', targetOption);
  (targetOption as HTMLElement).focus();
  setFocusedOption && setFocusedOption(targetOption);
  targetOption.scrollIntoView({ block: 'center' });
};
