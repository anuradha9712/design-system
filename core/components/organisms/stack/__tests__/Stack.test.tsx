import * as React from 'react';
import { render, screen } from '@testing-library/react';
import { Stack, StackItem } from '@/index';
import { testHelper, filterUndefined, valueHelper, testMessageHelper } from '@/utils/testHelper';
import { StackItemProps as Props } from '@/index.type';

const BooleanValue = [true, false];
const listSize = ['standard', 'compressed', 'tight'];
const listType = ['option', 'description', 'resource'];
const nestedRowElement = <div>nested row</div>;
const children = <div>stack item element</div>;
const stackItemId = 'stack-item-id';

/*
==DONE===
disable padding
States 
custom padding class
show divider


===TODO===
custom class to stack
nested row-> expanded
stack item length
check for ul, li element type
draggable icon

 */

// const CustomStack = (props: StackItemProps) => {
//   return (
//     <Stack>
//       <StackItem {...props}>{children}</StackItem>
//     </Stack>
//   );
// };

describe('Stack component snapshot', () => {
  const mapper: Record<string, any> = {
    showDivider: valueHelper(BooleanValue, { required: true, iterate: true }),
    disabled: valueHelper(BooleanValue, { required: true, iterate: true }),
    disablePadding: valueHelper(BooleanValue, { required: true, iterate: true }),
    draggable: valueHelper(BooleanValue, { required: true, iterate: true }),
    size: valueHelper(listSize, { required: true, iterate: false }),
    type: valueHelper(listType, { required: true, iterate: false }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const { baseElement } = render(
        <Stack>
          <StackItem {...attr}>{children}</StackItem>
        </Stack>
      );
      expect(baseElement).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('Stack component snapshot for nested row', () => {
  const mapper: Record<string, any> = {
    expanded: valueHelper(BooleanValue, { required: true, iterate: true }),
    nestedRow: valueHelper(nestedRowElement, { required: true, iterate: false }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const { baseElement } = render(
        <Stack>
          <StackItem {...attr}>{children}</StackItem>
        </Stack>
      );
      expect(baseElement).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('Stack component snapshot for states', () => {
  const mapper: Record<string, any> = {
    selected: valueHelper(BooleanValue, { required: true, iterate: true }),
    activated: valueHelper(BooleanValue, { required: true, iterate: true }),
    type: valueHelper(listType, { required: true, iterate: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const { baseElement } = render(
        <Stack>
          <StackItem {...attr}>{children}</StackItem>
        </Stack>
      );
      expect(baseElement).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('Stack Item component test for list size: standard', () => {
  it('check for padding classes for size:standard when disablePadding:false', () => {
    const { getByTestId } = render(
      <Stack size="standard">
        <StackItem id={stackItemId}>{children}</StackItem>
      </Stack>
    );
    const stackItem = getByTestId('DesignSystem-Stack-ItemWrapper');
    expect(stackItem).toHaveClass('px-6 py-5');
  });

  it('check for padding classes for size:standard when disablePadding:true', () => {
    const { getByTestId } = render(
      <Stack size="standard">
        <StackItem id={stackItemId} disablePadding={true}>
          {children}
        </StackItem>
      </Stack>
    );
    const stackItem = getByTestId('DesignSystem-Stack-ItemWrapper');
    expect(stackItem).not.toHaveClass('px-6 py-5');
  });
});

describe('Stack Item component test for list size: tight', () => {
  it('check for padding classes for size:tight when disablePadding:false', () => {
    const { getByTestId } = render(
      <Stack size="tight">
        <StackItem id={stackItemId}>{children}</StackItem>
      </Stack>
    );
    const stackItem = getByTestId('DesignSystem-Stack-ItemWrapper');
    expect(stackItem).toHaveClass('px-6 py-3');
  });

  it('check for padding classes for size:tight when disablePadding:true', () => {
    const { getByTestId } = render(
      <Stack size="tight">
        <StackItem id={stackItemId} disablePadding={true}>
          {children}
        </StackItem>
      </Stack>
    );
    const stackItem = getByTestId('DesignSystem-Stack-ItemWrapper');
    expect(stackItem).not.toHaveClass('px-6 py-3');
  });
});

describe('Stack Item component test for list size: compressed', () => {
  it('check for padding classes for size:compressed when disablePadding:false', () => {
    const { getByTestId } = render(
      <Stack size="compressed">
        <StackItem id={stackItemId}>{children}</StackItem>
      </Stack>
    );
    const stackItem = getByTestId('DesignSystem-Stack-ItemWrapper');
    expect(stackItem).toHaveClass('px-6 py-4');
  });

  it('check for padding classes for size:compressed when disablePadding:true', () => {
    const { getByTestId } = render(
      <Stack size="compressed">
        <StackItem id={stackItemId} disablePadding={true}>
          {children}
        </StackItem>
      </Stack>
    );
    const stackItem = getByTestId('DesignSystem-Stack-ItemWrapper');
    expect(stackItem).not.toHaveClass('px-6 py-4');
  });
});

describe('Stack Item component test for list type: option', () => {
  it('check for classes for type:option', () => {
    const { getByTestId } = render(
      <Stack type="option">
        <StackItem id={stackItemId} selected={true}>
          {children}
        </StackItem>
      </Stack>
    );
    const stackItem = getByTestId('DesignSystem-Stack-ItemWrapper');
    expect(stackItem).toHaveClass('Stack-item--selected');
    expect(stackItem).toHaveClass('Stack-item');
  });
});

describe('Stack Item component test for list type: resource', () => {
  it('check for classes for type:resource', () => {
    const { getByTestId } = render(
      <Stack type="resource">
        <StackItem id={stackItemId} activated={true}>
          {children}
        </StackItem>
      </Stack>
    );
    const stackItem = getByTestId('DesignSystem-Stack-ItemWrapper');
    expect(stackItem).toHaveClass('Stack-item--activated');
    expect(stackItem).toHaveClass('Stack-item');
  });
});

describe('Stack Item component test for list type: description', () => {
  it('check for classes for type:description', () => {
    const { getByTestId } = render(
      <Stack type="description">
        <StackItem id={stackItemId}>{children}</StackItem>
      </Stack>
    );
    const stackItem = getByTestId('DesignSystem-Stack-ItemWrapper');
    expect(stackItem).not.toHaveClass('Stack-item');
  });
});

describe('Stack Item component test for disabled classes', () => {
  const { getByTestId } = render(
    <StackItem id={stackItemId} disabled={true}>
      {children}
    </StackItem>
  );
  const stackItem = getByTestId('DesignSystem-Stack-ItemWrapper');
  expect(stackItem).toHaveClass('Stack-item--disabled');
});

describe('Stack Item component test for custom classes', () => {
  it('check for custom class', () => {
    const { getByTestId } = render(
      <StackItem id={stackItemId} className="custom-class">
        {children}
      </StackItem>
    );
    const stackItem = getByTestId('DesignSystem-Stack-ItemWrapper');
    expect(stackItem).toHaveClass('custom-class');
  });

  it('check to override default padding classes', () => {
    const { getByTestId } = render(
      <StackItem id={stackItemId} className="p-0" disablePadding={true}>
        {children}
      </StackItem>
    );
    const stackItem = getByTestId('DesignSystem-Stack-ItemWrapper');
    expect(stackItem).toHaveClass('p-0');
  });
});

describe('Stack Item component test for divider', () => {
  it('check for divider present below list item', () => {
    const { getByTestId } = render(
      <Stack showDivider={true}>
        <StackItem id={stackItemId}>{children}</StackItem>
      </Stack>
    );
    const divider = getByTestId('DesignSystem-Divider');
    expect(divider).toBeInTheDocument();
  });

  it('check for divider not present below list item', () => {
    render(
      <Stack showDivider={false}>
        <StackItem id={stackItemId}>{children}</StackItem>
      </Stack>
    );
    const divider = screen.queryByText('DesignSystem-Divider');
    expect(divider).not.toBeInTheDocument();
  });
});
