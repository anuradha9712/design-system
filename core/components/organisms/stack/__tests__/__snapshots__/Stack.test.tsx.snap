// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stack component snapshot 
  showDivider: false, disabled: false, disablePadding: false, draggable: false, size: ["standard","compressed","tight"], type: ["option","description","resource"]
 1`] = `
<body>
  <div>
    <ul
      class="Stack"
      data-test="DesignSystem-Stack"
    >
      <li
        data-test="DesignSystem-Stack-Item"
        draggable="false"
      >
        <div
          class="px-6 Stack-item"
          data-test="DesignSystem-Stack-ItemWrapper"
          tabindex="0"
        >
          <div>
            stack item element
          </div>
        </div>
      </li>
    </ul>
  </div>
</body>
`;

exports[`Stack component snapshot 
  showDivider: false, disabled: false, disablePadding: false, draggable: true, size: ["standard","compressed","tight"], type: ["option","description","resource"]
 1`] = `
<body>
  <div>
    <ul
      class="Stack"
      data-test="DesignSystem-Stack"
    >
      <li
        data-test="DesignSystem-Stack-Item"
        draggable="true"
      >
        <div
          class="px-6 Stack-item"
          data-test="DesignSystem-Stack-ItemWrapper"
          tabindex="0"
        >
          <div>
            <i
              class="material-icons material-icons-round Icon"
              role="button"
              style="font-size: 16px; width: 16px;"
            >
              drag_indicator_round
            </i>
          </div>
          <div>
            stack item element
          </div>
        </div>
      </li>
    </ul>
  </div>
</body>
`;

exports[`Stack component snapshot 
  showDivider: false, disabled: false, disablePadding: true, draggable: false, size: ["standard","compressed","tight"], type: ["option","description","resource"]
 1`] = `
<body>
  <div>
    <ul
      class="Stack"
      data-test="DesignSystem-Stack"
    >
      <li
        data-test="DesignSystem-Stack-Item"
        draggable="false"
      >
        <div
          class="Stack-item"
          data-test="DesignSystem-Stack-ItemWrapper"
          tabindex="0"
        >
          <div>
            stack item element
          </div>
        </div>
      </li>
    </ul>
  </div>
</body>
`;

exports[`Stack component snapshot 
  showDivider: false, disabled: false, disablePadding: true, draggable: true, size: ["standard","compressed","tight"], type: ["option","description","resource"]
 1`] = `
<body>
  <div>
    <ul
      class="Stack"
      data-test="DesignSystem-Stack"
    >
      <li
        data-test="DesignSystem-Stack-Item"
        draggable="true"
      >
        <div
          class="Stack-item"
          data-test="DesignSystem-Stack-ItemWrapper"
          tabindex="0"
        >
          <div>
            <i
              class="material-icons material-icons-round Icon"
              role="button"
              style="font-size: 16px; width: 16px;"
            >
              drag_indicator_round
            </i>
          </div>
          <div>
            stack item element
          </div>
        </div>
      </li>
    </ul>
  </div>
</body>
`;

exports[`Stack component snapshot 
  showDivider: false, disabled: true, disablePadding: false, draggable: false, size: ["standard","compressed","tight"], type: ["option","description","resource"]
 1`] = `
<body>
  <div>
    <ul
      class="Stack"
      data-test="DesignSystem-Stack"
    >
      <li
        data-test="DesignSystem-Stack-Item"
        draggable="false"
      >
        <div
          class="px-6 Stack-item--disabled Stack-item"
          data-test="DesignSystem-Stack-ItemWrapper"
          tabindex="0"
        >
          <div>
            stack item element
          </div>
        </div>
      </li>
    </ul>
  </div>
</body>
`;

exports[`Stack component snapshot 
  showDivider: false, disabled: true, disablePadding: false, draggable: true, size: ["standard","compressed","tight"], type: ["option","description","resource"]
 1`] = `
<body>
  <div>
    <ul
      class="Stack"
      data-test="DesignSystem-Stack"
    >
      <li
        data-test="DesignSystem-Stack-Item"
        draggable="true"
      >
        <div
          class="px-6 Stack-item--disabled Stack-item"
          data-test="DesignSystem-Stack-ItemWrapper"
          tabindex="0"
        >
          <div>
            <i
              class="material-icons material-icons-round Icon"
              role="button"
              style="font-size: 16px; width: 16px;"
            >
              drag_indicator_round
            </i>
          </div>
          <div>
            stack item element
          </div>
        </div>
      </li>
    </ul>
  </div>
</body>
`;

exports[`Stack component snapshot 
  showDivider: false, disabled: true, disablePadding: true, draggable: false, size: ["standard","compressed","tight"], type: ["option","description","resource"]
 1`] = `
<body>
  <div>
    <ul
      class="Stack"
      data-test="DesignSystem-Stack"
    >
      <li
        data-test="DesignSystem-Stack-Item"
        draggable="false"
      >
        <div
          class="Stack-item--disabled Stack-item"
          data-test="DesignSystem-Stack-ItemWrapper"
          tabindex="0"
        >
          <div>
            stack item element
          </div>
        </div>
      </li>
    </ul>
  </div>
</body>
`;

exports[`Stack component snapshot 
  showDivider: false, disabled: true, disablePadding: true, draggable: true, size: ["standard","compressed","tight"], type: ["option","description","resource"]
 1`] = `
<body>
  <div>
    <ul
      class="Stack"
      data-test="DesignSystem-Stack"
    >
      <li
        data-test="DesignSystem-Stack-Item"
        draggable="true"
      >
        <div
          class="Stack-item--disabled Stack-item"
          data-test="DesignSystem-Stack-ItemWrapper"
          tabindex="0"
        >
          <div>
            <i
              class="material-icons material-icons-round Icon"
              role="button"
              style="font-size: 16px; width: 16px;"
            >
              drag_indicator_round
            </i>
          </div>
          <div>
            stack item element
          </div>
        </div>
      </li>
    </ul>
  </div>
</body>
`;

exports[`Stack component snapshot 
  showDivider: true, disabled: false, disablePadding: false, draggable: false, size: ["standard","compressed","tight"], type: ["option","description","resource"]
 1`] = `
<body>
  <div>
    <ul
      class="Stack"
      data-test="DesignSystem-Stack"
    >
      <li
        data-test="DesignSystem-Stack-Item"
        draggable="false"
      >
        <div
          class="px-6 Stack-item"
          data-test="DesignSystem-Stack-ItemWrapper"
          tabindex="0"
        >
          <div>
            stack item element
          </div>
        </div>
        <hr
          class="Divider Divider--horizontal Divider--basic"
          data-test="DesignSystem-Divider"
        />
      </li>
    </ul>
  </div>
</body>
`;

exports[`Stack component snapshot 
  showDivider: true, disabled: false, disablePadding: false, draggable: true, size: ["standard","compressed","tight"], type: ["option","description","resource"]
 1`] = `
<body>
  <div>
    <ul
      class="Stack"
      data-test="DesignSystem-Stack"
    >
      <li
        data-test="DesignSystem-Stack-Item"
        draggable="true"
      >
        <div
          class="px-6 Stack-item"
          data-test="DesignSystem-Stack-ItemWrapper"
          tabindex="0"
        >
          <div>
            <i
              class="material-icons material-icons-round Icon"
              role="button"
              style="font-size: 16px; width: 16px;"
            >
              drag_indicator_round
            </i>
          </div>
          <div>
            stack item element
          </div>
        </div>
        <hr
          class="Divider Divider--horizontal Divider--basic"
          data-test="DesignSystem-Divider"
        />
      </li>
    </ul>
  </div>
</body>
`;

exports[`Stack component snapshot 
  showDivider: true, disabled: false, disablePadding: true, draggable: false, size: ["standard","compressed","tight"], type: ["option","description","resource"]
 1`] = `
<body>
  <div>
    <ul
      class="Stack"
      data-test="DesignSystem-Stack"
    >
      <li
        data-test="DesignSystem-Stack-Item"
        draggable="false"
      >
        <div
          class="Stack-item"
          data-test="DesignSystem-Stack-ItemWrapper"
          tabindex="0"
        >
          <div>
            stack item element
          </div>
        </div>
        <hr
          class="Divider Divider--horizontal Divider--basic"
          data-test="DesignSystem-Divider"
        />
      </li>
    </ul>
  </div>
</body>
`;

exports[`Stack component snapshot 
  showDivider: true, disabled: false, disablePadding: true, draggable: true, size: ["standard","compressed","tight"], type: ["option","description","resource"]
 1`] = `
<body>
  <div>
    <ul
      class="Stack"
      data-test="DesignSystem-Stack"
    >
      <li
        data-test="DesignSystem-Stack-Item"
        draggable="true"
      >
        <div
          class="Stack-item"
          data-test="DesignSystem-Stack-ItemWrapper"
          tabindex="0"
        >
          <div>
            <i
              class="material-icons material-icons-round Icon"
              role="button"
              style="font-size: 16px; width: 16px;"
            >
              drag_indicator_round
            </i>
          </div>
          <div>
            stack item element
          </div>
        </div>
        <hr
          class="Divider Divider--horizontal Divider--basic"
          data-test="DesignSystem-Divider"
        />
      </li>
    </ul>
  </div>
</body>
`;

exports[`Stack component snapshot 
  showDivider: true, disabled: true, disablePadding: false, draggable: false, size: ["standard","compressed","tight"], type: ["option","description","resource"]
 1`] = `
<body>
  <div>
    <ul
      class="Stack"
      data-test="DesignSystem-Stack"
    >
      <li
        data-test="DesignSystem-Stack-Item"
        draggable="false"
      >
        <div
          class="px-6 Stack-item--disabled Stack-item"
          data-test="DesignSystem-Stack-ItemWrapper"
          tabindex="0"
        >
          <div>
            stack item element
          </div>
        </div>
        <hr
          class="Divider Divider--horizontal Divider--basic"
          data-test="DesignSystem-Divider"
        />
      </li>
    </ul>
  </div>
</body>
`;

exports[`Stack component snapshot 
  showDivider: true, disabled: true, disablePadding: false, draggable: true, size: ["standard","compressed","tight"], type: ["option","description","resource"]
 1`] = `
<body>
  <div>
    <ul
      class="Stack"
      data-test="DesignSystem-Stack"
    >
      <li
        data-test="DesignSystem-Stack-Item"
        draggable="true"
      >
        <div
          class="px-6 Stack-item--disabled Stack-item"
          data-test="DesignSystem-Stack-ItemWrapper"
          tabindex="0"
        >
          <div>
            <i
              class="material-icons material-icons-round Icon"
              role="button"
              style="font-size: 16px; width: 16px;"
            >
              drag_indicator_round
            </i>
          </div>
          <div>
            stack item element
          </div>
        </div>
        <hr
          class="Divider Divider--horizontal Divider--basic"
          data-test="DesignSystem-Divider"
        />
      </li>
    </ul>
  </div>
</body>
`;

exports[`Stack component snapshot 
  showDivider: true, disabled: true, disablePadding: true, draggable: false, size: ["standard","compressed","tight"], type: ["option","description","resource"]
 1`] = `
<body>
  <div>
    <ul
      class="Stack"
      data-test="DesignSystem-Stack"
    >
      <li
        data-test="DesignSystem-Stack-Item"
        draggable="false"
      >
        <div
          class="Stack-item--disabled Stack-item"
          data-test="DesignSystem-Stack-ItemWrapper"
          tabindex="0"
        >
          <div>
            stack item element
          </div>
        </div>
        <hr
          class="Divider Divider--horizontal Divider--basic"
          data-test="DesignSystem-Divider"
        />
      </li>
    </ul>
  </div>
</body>
`;

exports[`Stack component snapshot 
  showDivider: true, disabled: true, disablePadding: true, draggable: true, size: ["standard","compressed","tight"], type: ["option","description","resource"]
 1`] = `
<body>
  <div>
    <li
      data-test="DesignSystem-Stack-Item"
      draggable="false"
      id="stack-item-id"
    >
      <div
        class="px-6 Stack-item--disabled Stack-item py-5"
        data-test="DesignSystem-Stack-ItemWrapper"
        tabindex="0"
      >
        <div>
          stack item element
        </div>
      </div>
      <hr
        class="Divider Divider--horizontal Divider--basic"
        data-test="DesignSystem-Divider"
      />
    </li>
  </div>
  <div>
    <ul
      class="Stack"
      data-test="DesignSystem-Stack"
    >
      <li
        data-test="DesignSystem-Stack-Item"
        draggable="true"
      >
        <div
          class="Stack-item--disabled Stack-item"
          data-test="DesignSystem-Stack-ItemWrapper"
          tabindex="0"
        >
          <div>
            <i
              class="material-icons material-icons-round Icon"
              role="button"
              style="font-size: 16px; width: 16px;"
            >
              drag_indicator_round
            </i>
          </div>
          <div>
            stack item element
          </div>
        </div>
        <hr
          class="Divider Divider--horizontal Divider--basic"
          data-test="DesignSystem-Divider"
        />
      </li>
    </ul>
  </div>
</body>
`;

exports[`Stack component snapshot for nested row 
  expanded: false, nestedRow: "[ReactNode]"
 1`] = `
<body>
  <div>
    <ul
      class="Stack"
      data-test="DesignSystem-Stack"
    >
      <li
        data-test="DesignSystem-Stack-Item"
        draggable="false"
      >
        <div
          class="px-6 Stack-item py-5"
          data-test="DesignSystem-Stack-ItemWrapper"
          tabindex="0"
        >
          <div>
            stack item element
          </div>
        </div>
        <hr
          class="Divider Divider--horizontal Divider--basic"
          data-test="DesignSystem-Divider"
        />
      </li>
    </ul>
  </div>
</body>
`;

exports[`Stack component snapshot for nested row 
  expanded: true, nestedRow: "[ReactNode]"
 1`] = `
<body>
  <div>
    <ul
      class="Stack"
      data-test="DesignSystem-Stack"
    >
      <li
        data-test="DesignSystem-Stack-Item"
        draggable="false"
      >
        <div
          class="px-6 Stack-item py-5"
          data-test="DesignSystem-Stack-ItemWrapper"
          tabindex="0"
        >
          <div>
            stack item element
          </div>
        </div>
        <div
          data-test="DesignSystem-Stack--Nested-Item"
        >
          <div>
            nested row
          </div>
        </div>
        <hr
          class="Divider Divider--horizontal Divider--basic"
          data-test="DesignSystem-Divider"
        />
      </li>
    </ul>
  </div>
</body>
`;

exports[`Stack component snapshot for states 
  selected: false, activated: false, type: "description"
 1`] = `
<body>
  <div>
    <ul
      class="Stack"
      data-test="DesignSystem-Stack"
    >
      <li
        data-test="DesignSystem-Stack-Item"
        draggable="false"
      >
        <div
          class="px-6 py-5"
          data-test="DesignSystem-Stack-ItemWrapper"
          tabindex="0"
        >
          <div>
            stack item element
          </div>
        </div>
        <hr
          class="Divider Divider--horizontal Divider--basic"
          data-test="DesignSystem-Divider"
        />
      </li>
    </ul>
  </div>
</body>
`;

exports[`Stack component snapshot for states 
  selected: false, activated: false, type: "option"
 1`] = `
<body>
  <div>
    <ul
      class="Stack"
      data-test="DesignSystem-Stack"
    >
      <li
        data-test="DesignSystem-Stack-Item"
        draggable="false"
      >
        <div
          class="px-6 Stack-item py-5"
          data-test="DesignSystem-Stack-ItemWrapper"
          tabindex="0"
        >
          <div>
            stack item element
          </div>
        </div>
        <hr
          class="Divider Divider--horizontal Divider--basic"
          data-test="DesignSystem-Divider"
        />
      </li>
    </ul>
  </div>
</body>
`;

exports[`Stack component snapshot for states 
  selected: false, activated: false, type: "resource"
 1`] = `
<body>
  <div>
    <ul
      class="Stack"
      data-test="DesignSystem-Stack"
    >
      <li
        data-test="DesignSystem-Stack-Item"
        draggable="false"
      >
        <div
          class="px-6 Stack-item py-5"
          data-test="DesignSystem-Stack-ItemWrapper"
          tabindex="0"
        >
          <div>
            stack item element
          </div>
        </div>
        <hr
          class="Divider Divider--horizontal Divider--basic"
          data-test="DesignSystem-Divider"
        />
      </li>
    </ul>
  </div>
</body>
`;

exports[`Stack component snapshot for states 
  selected: false, activated: true, type: "description"
 1`] = `
<body>
  <div>
    <ul
      class="Stack"
      data-test="DesignSystem-Stack"
    >
      <li
        data-test="DesignSystem-Stack-Item"
        draggable="false"
      >
        <div
          class="px-6 py-5"
          data-test="DesignSystem-Stack-ItemWrapper"
          tabindex="0"
        >
          <div>
            stack item element
          </div>
        </div>
        <hr
          class="Divider Divider--horizontal Divider--basic"
          data-test="DesignSystem-Divider"
        />
      </li>
    </ul>
  </div>
</body>
`;

exports[`Stack component snapshot for states 
  selected: false, activated: true, type: "option"
 1`] = `
<body>
  <div>
    <ul
      class="Stack"
      data-test="DesignSystem-Stack"
    >
      <li
        data-test="DesignSystem-Stack-Item"
        draggable="false"
      >
        <div
          class="px-6 Stack-item py-5"
          data-test="DesignSystem-Stack-ItemWrapper"
          tabindex="0"
        >
          <div>
            stack item element
          </div>
        </div>
        <hr
          class="Divider Divider--horizontal Divider--basic"
          data-test="DesignSystem-Divider"
        />
      </li>
    </ul>
  </div>
</body>
`;

exports[`Stack component snapshot for states 
  selected: false, activated: true, type: "resource"
 1`] = `
<body>
  <div>
    <ul
      class="Stack"
      data-test="DesignSystem-Stack"
    >
      <li
        data-test="DesignSystem-Stack-Item"
        draggable="false"
      >
        <div
          class="px-6 Stack-item Stack-item--activated py-5"
          data-test="DesignSystem-Stack-ItemWrapper"
          tabindex="0"
        >
          <div>
            stack item element
          </div>
        </div>
        <hr
          class="Divider Divider--horizontal Divider--basic"
          data-test="DesignSystem-Divider"
        />
      </li>
    </ul>
  </div>
</body>
`;

exports[`Stack component snapshot for states 
  selected: true, activated: false, type: "description"
 1`] = `
<body>
  <div>
    <ul
      class="Stack"
      data-test="DesignSystem-Stack"
    >
      <li
        data-test="DesignSystem-Stack-Item"
        draggable="false"
      >
        <div
          class="px-6 py-5"
          data-test="DesignSystem-Stack-ItemWrapper"
          tabindex="0"
        >
          <div>
            stack item element
          </div>
        </div>
        <hr
          class="Divider Divider--horizontal Divider--basic"
          data-test="DesignSystem-Divider"
        />
      </li>
    </ul>
  </div>
</body>
`;

exports[`Stack component snapshot for states 
  selected: true, activated: false, type: "option"
 1`] = `
<body>
  <div>
    <ul
      class="Stack"
      data-test="DesignSystem-Stack"
    >
      <li
        data-test="DesignSystem-Stack-Item"
        draggable="false"
      >
        <div
          class="px-6 Stack-item Stack-item--selected py-5"
          data-test="DesignSystem-Stack-ItemWrapper"
          tabindex="0"
        >
          <div>
            stack item element
          </div>
        </div>
        <hr
          class="Divider Divider--horizontal Divider--basic"
          data-test="DesignSystem-Divider"
        />
      </li>
    </ul>
  </div>
</body>
`;

exports[`Stack component snapshot for states 
  selected: true, activated: false, type: "resource"
 1`] = `
<body>
  <div>
    <ul
      class="Stack"
      data-test="DesignSystem-Stack"
    >
      <li
        data-test="DesignSystem-Stack-Item"
        draggable="false"
      >
        <div
          class="px-6 Stack-item py-5"
          data-test="DesignSystem-Stack-ItemWrapper"
          tabindex="0"
        >
          <div>
            stack item element
          </div>
        </div>
        <hr
          class="Divider Divider--horizontal Divider--basic"
          data-test="DesignSystem-Divider"
        />
      </li>
    </ul>
  </div>
</body>
`;

exports[`Stack component snapshot for states 
  selected: true, activated: true, type: "description"
 1`] = `
<body>
  <div>
    <ul
      class="Stack"
      data-test="DesignSystem-Stack"
    >
      <li
        data-test="DesignSystem-Stack-Item"
        draggable="false"
      >
        <div
          class="px-6 py-5"
          data-test="DesignSystem-Stack-ItemWrapper"
          tabindex="0"
        >
          <div>
            stack item element
          </div>
        </div>
        <hr
          class="Divider Divider--horizontal Divider--basic"
          data-test="DesignSystem-Divider"
        />
      </li>
    </ul>
  </div>
</body>
`;

exports[`Stack component snapshot for states 
  selected: true, activated: true, type: "option"
 1`] = `
<body>
  <div>
    <ul
      class="Stack"
      data-test="DesignSystem-Stack"
    >
      <li
        data-test="DesignSystem-Stack-Item"
        draggable="false"
      >
        <div
          class="px-6 Stack-item Stack-item--selected py-5"
          data-test="DesignSystem-Stack-ItemWrapper"
          tabindex="0"
        >
          <div>
            stack item element
          </div>
        </div>
        <hr
          class="Divider Divider--horizontal Divider--basic"
          data-test="DesignSystem-Divider"
        />
      </li>
    </ul>
  </div>
</body>
`;

exports[`Stack component snapshot for states 
  selected: true, activated: true, type: "resource"
 1`] = `
<body>
  <div>
    <ul
      class="Stack"
      data-test="DesignSystem-Stack"
    >
      <li
        data-test="DesignSystem-Stack-Item"
        draggable="false"
      >
        <div
          class="px-6 Stack-item Stack-item--activated py-5"
          data-test="DesignSystem-Stack-ItemWrapper"
          tabindex="0"
        >
          <div>
            stack item element
          </div>
        </div>
        <hr
          class="Divider Divider--horizontal Divider--basic"
          data-test="DesignSystem-Divider"
        />
      </li>
    </ul>
  </div>
</body>
`;
